src/syntaxtree.c

```diff
@@ -65,6 +65,9 @@ static const CompilerT Compilers[];
     }
 #define ARG_(name) ARG(name, SyntaxTreeCompiler)
 
+// use ARGS as last "arg" which collect all the following args together
```

Comment:
I've read this comment a couple of times, and I don't know what it means

Reply:
Indeed, that was a leftover, I've now adapted it to the commenting style of the rest of the code in this file (i.e., I removed it). I plan to add some proper comments to this file, explaining how everything works, at a future point, now that @sebasguts and me worked it out (we think).

Reply:
Main reason that I am not adding more comments right now, besides it being late, is that @sebasguts is working on a PR to add code that converts the output of the "SyntraxTree" functions back into GAP functions, and that code is based on this PR, so I'd prefer not to modify it to heavily, at least not until I have a chance to sync it with @sebasguts work.